backend-rust/src/graphql_api/reward_metrics.rs

```diff
@@ -0,0 +1,102 @@
+use crate::{
+    graphql_api::{get_pool, ApiError, ApiResult, MetricsPeriod},
+    scalar_types::{DateTime, TimeSpan},
+};
+use async_graphql::{types, Context, Object, SimpleObject};
+use chrono::Utc;
+use sqlx::{postgres::types::PgInterval, PgPool};
+use std::sync::Arc;
+
+#[derive(Default)]
+pub(crate) struct QueryRewardMetrics;
+
+#[Object]
+impl QueryRewardMetrics {
+    async fn reward_metrics<'a>(
+        &self,
+        ctx: &Context<'a>,
+        period: MetricsPeriod,
+    ) -> ApiResult<RewardMetrics> {
+        reward_metrics(period, None, get_pool(ctx)?).await
+    }
+
+    async fn reward_metrics_for_account<'a>(
+        &self,
+        ctx: &Context<'a>,
+        period: MetricsPeriod,
+        account_id: types::ID,
+    ) -> ApiResult<RewardMetrics> {
+        reward_metrics(period, Some(account_id), get_pool(ctx)?).await
+    }
+}
+
+async fn reward_metrics(
+    period: MetricsPeriod,
+    account_id: Option<types::ID>,
+    pool: &PgPool,
+) -> ApiResult<RewardMetrics> {
+    let end_time = Utc::now();
+    let before_time = end_time - period.as_duration();
+    let bucket_width = period.bucket_width();
+
+    let bucket_interval: PgInterval =
+        bucket_width.try_into().map_err(|err| ApiError::DurationOutOfRange(Arc::new(err)))?;
+
+    let value: Option<i64> =
+        account_id.map(|x| x.try_into().map_err(ApiError::InvalidIdInt)).transpose()?;
+
+    let rows = sqlx::query!(
+        r#"
+        SELECT
+            bucket_time.bucket_start AS "bucket_time!",
+            (SELECT COALESCE(SUM(amount), 0)
+                            FROM metrics_rewards
+                            WHERE block_slot_time BETWEEN bucket_time.bucket_start AND bucket_time.bucket_end
+                            AND ($4::BIGINT IS NULL OR account_index = $4::BIGINT))::BIGINT AS "accumulated_amount!"
```

Comment:
Has been tested against stagenet which by far has the most dense dataset on rewards. Performed fine.